### DISCUSSION TITLE
Library view and navigating dependencies

### DISCUSSION THREAD

**Discussion by dphaldes** [nvim-java/nvim-java#78] [General]
(This might be out of scope for this plugin specifically)

I often find myself looking through classes and file structure of my dependencies to look at the API or figure out how specific things are done in provided examples and library classes (sources). It would be really helpful if there was a way to look at all dependencies/libraries at a glance and traverse through them in some kind of tree (maybe even an extension to neo-tree).

How do you tackle such a situation? Could we have a system in place for `nvim-java` to look through the jars? Or should a generalized approach through plugin is the way to go ?

**Comment by dphaldes** [comment]
Found some projects that might be helpful https://github.com/uzxmx/coc-java-dependency and https://github.com/Microsoft/vscode-java-dependency. The latter one is a jdtls extension, and hopefully we can just plug it into `nvim-java`


Edit : Other related projects that might also be helpful https://github.com/microsoft/vscode-gradle & https://github.com/Microsoft/vscode-maven. I couldn't see specific classes/files being accessed through these, but be helpful for implementation or something

**Reply by s1n7ax** [reply]
Thanks. This is not possible right now but adding this to the backlog. It should be possible to use neo-tree as the view. You can view them here but this will not be implemented soon
https://github.com/orgs/nvim-java/projects/1/views/2

**Reply by PiotrRaszkowski** [reply]
https://github.com/JavaHello/java-deps.nvim could be the option... but...

they suggest:
local function jdtls_on_attach(client, bufnr)
    local root_dir = require('jdtls').setup.find_root(root_files)
    require("java-deps").attach(client, bufnr, root_dir)

I don't know how to get the root_dir in nvim-java :(
